src/plugins/pendo/visitorData.ts

```diff
@@ -0,0 +1,100 @@
+import forEach from 'lodash/forEach'
+import { logger } from 'lib/logger'
+
+const moduleLogger = logger.child({ namespace: ['Plugin', 'Pendo', 'VisitorDataManager'] })
+
+export type VisitorId = {
+  id: string
+  expiry: number
+}
+
+export type VisitorData = {
+  visitorId: VisitorId
+  consent: Record<string, true>
+}
+
+export class VisitorDataManager {
+  static readonly #consentMap = new Map<string, boolean>()
+  static #visitorId: VisitorId = { id: '', expiry: 0 }
+
+  static load(): void {
+    const raw = window.localStorage.getItem('visitorData') ?? '{"consent":{}}'
```

Comment:
more of a question than anything, but can't local storage be accessed by anything running on your page (not secure), yet we are pulling the "unsecure" data and assigning to private `#` field as if it is secure or truly protected/not intended to be leaked.

is my understanding here correct, or is is marginally more secure than exposing on the class object itself so it is a slight security improvement?

doesn't ultimately hurt either way, just curious

Reply:
I understand what your saying. I had to rework some of MrNerdHairs code because it was overly complicated. This code could benefit from another pass of simplification. 

In fact a lot of this code could probably be significantly simplified. 

I'd prefer to leave that for a future ticket to avoid more delays right now.

Reply:
Yeah totally fair, just wanted to bring it up as a talking point